crates/pop-parachains/src/bench/mod.rs

```diff
@@ -0,0 +1,238 @@
+// SPDX-License-Identifier: GPL-3.0
+
+use clap::Parser;
+use frame_benchmarking_cli::PalletCmd;
+use sc_chain_spec::GenesisConfigBuilderRuntimeCaller;
+use sp_runtime::traits::BlakeTwo256;
+use std::{
+	collections::HashMap,
+	fmt::Display,
+	fs::{self, File},
+	io::Read,
+	path::{Path, PathBuf},
+	process::{Child, Command, Stdio},
+};
+use strum_macros::{EnumIter, EnumMessage as EnumMessageDerive};
+use tempfile::NamedTempFile;
+
+/// Provides functionality for sourcing binaries of the benchmarking CLI.
+pub mod binary;
+
+/// The default `development` preset used to communicate with the runtime via
+/// [`GenesisBuilder`] interface.
+///
+/// (Recommended for testing with a single node, e.g., for benchmarking)
+pub const GENESIS_BUILDER_DEV_PRESET: &str = "development";
+
+type HostFunctions = (
+	sp_statement_store::runtime_api::HostFunctions,
+	cumulus_primitives_proof_size_hostfunction::storage_proof_size::HostFunctions,
+);
+
+/// Type alias for records where the key is the pallet name and the value is a array of its
+/// extrinsics.
+pub type PalletExtrinsicsRegistry = HashMap<String, Vec<String>>;
+
+/// How the genesis state for benchmarking should be built.
+#[derive(clap::ValueEnum, Debug, Eq, PartialEq, Clone, Copy, EnumIter, EnumMessageDerive)]
+#[clap(rename_all = "kebab-case")]
+pub enum GenesisBuilderPolicy {
+	/// Do not provide any genesis state.
+	None,
+	/// Let the runtime build the genesis state through its `BuildGenesisConfig` runtime API.
+	Runtime,
+}
+
+impl Display for GenesisBuilderPolicy {
+	fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+		let s = match self {
+			GenesisBuilderPolicy::None => "none",
+			GenesisBuilderPolicy::Runtime => "runtime",
+		};
+		write!(f, "{}", s)
+	}
+}
+
+impl From<String> for GenesisBuilderPolicy {
+	fn from(s: String) -> Self {
+		match s {
+			s if s == *"none" => GenesisBuilderPolicy::None,
+			s if s == *"runtime" => GenesisBuilderPolicy::Runtime,
+			_ => unreachable!(),
+		}
+	}
+}
+
+/// Get genesis builder preset names of the runtime.
+///
+/// # Arguments
+/// * `binary_path` - Path to the runtime WASM binary.
```

Comment:
Suggesting this removal so that when the time comes where runtimes will be ran on PVM we don't have to make this change:
```suggestion
/// * `binary_path` - Path to the runtime binary.
```

Reply:
Resolved in https://github.com/r0gue-io/pop-cli/pull/420/commits/40a80c6c0bcd771ef886c3e5e84b67cc6e9634df